### DISCUSSION TITLE
listener - reportlabels v pravidlech

### DISCUSSION THREAD

**Discussion by onhru** [canarytrace/documentation#206] [Listener]
Hezky den,

pro nastaveni pravidel pro listener se pouzivaji reportlabels, tak aby ve zprave bylo vice informaci. pro posilani emailu se toto nastaveni nebere v potaz protoze do tela emailu se vzdy napise pouze:
monitoring production uuid

monitoring production uuid:9bc33118d3b1074ab24677c11a9d0afd

vypada to ze se pro email nebere toto nastaveni

s pranim hezkeho dne

Ondrej Hrusa - direct-auto

**Comment by rdpanek** [comment]
Dobrý den @onhru, omlouvám se za chybné vysvětlení funkce `reportLabels`.

Tato vlastnost v pravidlech neslouží k definování doplňkových informací v reportech, ale jako selektory labelů, které se pak mají objevit v `c.listener.events-*`.

Do tohoto indexu se ukládají výsledky analýzy pravidel, včetně dokumentů, které konkrétní pravidlo porušily. Vzhledem k tomu, že listener vyhodnocuje více indexů podle pravidel, musí existovat mechanismus, který do events zapíše dodatečné informace z porušeného pravidla, nalezeného v konkrétním indexu a dokumentu.

Když se podíváte na obrázky, tak v `reportLabels` jsou názvy labelů z konkrétního indexu, které umožňují obohatit pole documents v indexu `c.listener.events-*` o další informace z konkrétního dokumentu, porušujícího pravidlo - viz druhý obrázek.

`reportLabels` nemůžete uvádět libovolné, ale pouze ty, které jsou obsaženy v daném indexu, na který je pravidlo aplikováno. Například v tomto případě, jak je na obrázku, můžeme do `reportLabels` dát pouze ty labely, které obsahuje index `c.audit`, a nemůžeme použít labely z jiného indexu, pokud se konkrétní pravidlo nevztahuje na jiný index.

Děkuji za podnět, doplníme dokumentaci.

<img width="1359" alt="Snímek obrazovky 2023-11-29 v 19 19 39" src="https://github.com/canarytrace/documentation/assets/230124/2a624dcc-0a80-40b5-a02f-b83493ae6f5f">
<img width="1704" alt="Snímek obrazovky 2023-11-29 v 19 19 51" src="https://github.com/canarytrace/documentation/assets/230124/131ac3a1-2eed-4db0-b24a-912842fd7e68">
